test/unit/specs/components/Personalization/createApplyPropositions.spec.js

```diff
@@ -154,96 +196,75 @@ describe("Personalization::createApplyPropositions", () => {
+    const executeDecisionsPromise = {
+      then: callback => callback(MIXED_PROPOSITIONS)
+    };
+    executeDecisions.and.returnValue(executeDecisionsPromise);
+
     const applyPropositions = createApplyPropositions({
-      executeDecisions,
-      showContainers
+      executeDecisions
     });
 
-    const metadata = {
-      randomScope: {
-        selector: "#home-item1",
-        actionType: "setHtml"
-      }
-    };
-
     return applyPropositions({
-      propositions: MIXED_PROPOSITIONS,
-      metadata
-    }).then(() => {
-      const executedPropositions = executeDecisions.calls.all()[0].args[0];
-      expect(executedPropositions.length).toEqual(3);
-      executedPropositions.forEach(proposition => {
+      propositions: MIXED_PROPOSITIONS
+    }).then(result => {
+      expect(result.propositions.length).toEqual(2);
+      result.propositions.forEach(proposition => {
         expect(proposition.scope).toEqual("home");
-        expect(proposition.items.length).toEqual(1);
-        if (proposition.items[0].id === "442358") {
-          expect(proposition.items[0].data.selector).toEqual("#root");
-          expect(proposition.items[0].data.type).toEqual("click");
-        } else if (proposition.items[0].id === "442359") {
-          expect(proposition.items[0].data.selector).toEqual("head");
-          expect(proposition.items[0].data.type).toEqual("appendHtml");
-        } else if (proposition.items[0].id === "442360") {
-          expect(proposition.items[0].data.selector).toBeUndefined();
-          expect(proposition.items[0].data.type).toBeUndefined();
-        }
+        expect(proposition.renderAttempted).toBeTrue();
       });
-      expect(executeDecisions).toHaveBeenCalledTimes(1);
-      expect(showContainers).toHaveBeenCalled();
     });
   });
 
-  it("it should set renderAttempted = true for all user-provided propositions", () => {
+  it("it should ignore items with unsupported schemas", () => {
+    const executeDecisionsPromise = {
+      then: callback => callback(MIXED_PROPOSITIONS)
+    };
+    executeDecisions.and.returnValue(executeDecisionsPromise);
+
+    const expectedItemIds = ["442358", "442359"];
+
     const applyPropositions = createApplyPropositions({
-      executeDecisions,
-      showContainers
+      executeDecisions
     });
 
     return applyPropositions({
       propositions: MIXED_PROPOSITIONS
-    }).then(result => {
-      expect(result.propositions.length).toEqual(3);
-      result.propositions.forEach(proposition => {
-        expect(proposition.renderAttempted).toBeTrue();
+    }).then(() => {
+      const executedPropositions = executeDecisions.calls.all()[0].args[0];
+      expect(executedPropositions.length).toEqual(2);
+      executedPropositions.forEach(proposition => {
+        expect(proposition.items.length).toEqual(1);
+        proposition.items.forEach(item => {
+          expect(expectedItemIds.includes(item.id));
+        });
       });
     });
   });
```

Comment:
Can you add a test to make sure that the original propositions array or underlying proposition objects aren't mutated when the metadata scopes are applied?